massa-bootstrap/src/server.rs

```diff
@@ -98,19 +102,16 @@ impl BootstrapManager {
 }
 
 /// See module level documentation for details
-pub async fn start_bootstrap_server(
+pub fn start_bootstrap_server<D: Duplex>(
     consensus_controller: Box<dyn ConsensusController>,
     network_command_sender: NetworkCommandSender,
     final_state: Arc<RwLock<FinalState>>,
     config: BootstrapConfig,
-    mut establisher: impl BSEstablisher,
+    listener: impl BSListener + Send + 'static,
     keypair: KeyPair,
     version: Version,
-) -> Result<Option<BootstrapManager>, Box<BootstrapError>> {
+) -> Result<Option<BootstrapManager<TcpStream>>, Box<BootstrapError>> {
     massa_trace!("bootstrap.lib.start_bootstrap_server", {});
-    let Some(listen_addr) = config.listen_addr else {
```

Comment:
we should keep that no ?

Reply:
it's changed from an establisher (with a `get_listener` call further down) to just passing in a listener.

listen address is no longer needed, because it was only used to get the listener. This now happens before you call the bootstrap start

Reply:
This change here can provide additional context https://github.com/massalabs/massa/pull/3685/commits/840cfa5d0346c2ca083ea5556e436e81c375ec9f